layers.go

```diff
@@ -416,6 +416,16 @@ func (r *layerStore) loadMounts() error {
 }
 
 func (r *layerStore) Save() error {
+	r.mountsLockfile.Lock()
+	defer r.mountsLockfile.Unlock()
+	defer r.mountsLockfile.Touch()
+	if err := r.saveLayers(); err != nil {
```

Comment:
We could move the call to `saveLayers()` before acquiring the mountsLockfile (i.e., to line 419). It would  further shorten the critical section.